### PR TITLE
detect-filesize: Convert unittests to new FAIL/PASS API

### PR DESCRIPTION
**Pull request by hsadia538**
- [X] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Previous PR: #8070
Link to [redmine](https://redmine.openinfosecfoundation.org/issues/4039) ticket:

Describe changes:

-Update detect-filesize unit tests to FAIL/PASS APIs

### CODE REVIEW

File: src/detect-filesize.c [hunk 1]
```diff
@@ -257,36 +249,28 @@ static int DetectFilesizeInitTest(
...
 
     *de_ctx = DetectEngineCtxInit();
-    if (*de_ctx == NULL) {
-        goto end;
-    }
+    FAIL_IF_NULL(*de_ctx);
 
     (*de_ctx)->flags |= DE_QUIET;
 
     (*de_ctx)->sig_list = SigInit(*de_ctx, fullstr);
```
Comment (victorjulien) [inline src/detect-filesize.c:266]: could you switch this to use `DetectEngineAppendSig`?

File: src/detect-filesize.c [hunk 2]
```diff
@@ -305,26 +289,19 @@ static int DetectFilesizeSetpTest01(void)
...
+    FAIL_IF_NULL(fsd);
+    FAIL_IF_NOT(fsd->arg1 == 1);
+    FAIL_IF_NOT(fsd->arg2 == 3);
+    FAIL_IF_NOT(fsd->mode == DETECT_UINT_RA);
 
-cleanup:
-    if (fsd)
-        DetectFilesizeFree(NULL, fsd);
+    DetectFilesizeFree(NULL, fsd);
     SigGroupCleanup(de_ctx);
```
Comment (victorjulien) [inline src/detect-filesize.c:300]: just `DetectEngineCtxFree` is enough here, so `SigGroupCleanup` and `SigCleanSignatures` can be removed

**Review by victorjulien — COMMENTED** [review]
Looking good. I have 2 minor suggestions inline.

File: src/detect-filesize.c (hunk 1 above)
Comment (hsadia538) [reply src/detect-filesize.c:266]: Would it be like this:
` FAIL_IF_NULL(DetectEngineAppendSig(de_ctx,"alert http any any -> any any (msg:\"Filesize " "test\"; filesize:%s; sid:1;)");`

Comment (jufajardini) [reply src/detect-filesize.c:266]: Can't it be `FAIL_IF_NULL(DetectEngineAppendSig(de_ctx, fullstr)`?

**Comment by hsadia538** [comment]
Followed by #8091

**Pull request closed without merging**